js/app.js

```diff
@@ -119,6 +138,47 @@ var handleClickOnItem = function(event){
 
 
 
-//----------------------------------Executable Code---------------------------------//
+//-------------------------------------Executable Code------------------------------------//
 renderItems();
-mallParent.addEventListener('click', handleClickOnItem);
+mallParent.addEventListener('click', handleClickOnItem, true);
+//----------------------------Chart JS & THML Canvas Lab----------------------------------//
+
+function renderChart() {
+    var labelData = [];
+    var clickData = [];
+    var viewData = [];
+    for (var i = 0; i < MallItem.allItems.length; i++) {
+      labelData.push(MallItem.allItems[i].name);
+      clickData.push(MallItem.allItems[i].clicked);
+      viewData.push(MallItem.allItems[i].views);
+    }
+  
+    var ctx = document.getElementById('my-chart').getContext('2d');
+  
+    new Chart(ctx, {
+      type: 'bar',
+      data: {
+        labels: labelData,
+        datasets: [{
+          label: '# of Clicks',
+          data: clickData,
+          backgroundColor: ['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange','Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange','Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange'],
+        }, {
+          label: '# of Views',
+          data: viewData,
+          backgroundColor: ['Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange','Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange','Red', 'Blue', 'Yellow', 'Green', 'Purple', 'Orange'],
+        }]
+      },
+      options: {
+        scales: {
+          yAxes: [{
+            ticks: {
+              beginAtZero: true
+            }
+          }]
+        }
+      }
+    })
+  }
+  
+  renderChart();
```

Comment:
I recommend rendering chart after 25 clicks. Right now, it is only showing data for your first 3 images viewed and that is it.